tools/codegen/gen.py

```diff
@@ -837,9 +847,55 @@ def write_with_template(self, filename: str, template_fn: str,
                 assert_never(env)
 
 
-    def write(self, filename: str, env_callable: Callable[[], Union[str, Union[str, Dict[str, object]]]]) -> None:
+    def write(self, filename: str, env_callable: Callable[[], Union[str, Union[str, Dict[str, Any]]]]) -> None:
         self.write_with_template(filename, filename, env_callable)
 
+    def write_sharded(
+            self,
+            filename: str,
+            items: List[T],
+            *,
+            key_fn: Callable[[T], str],
+            env_callable: Callable[[T], Dict[str, List[str]]],
+            num_shards: int,
+            base_env: Dict[str, Any] = {},
```

Comment:
I think the empty dict here is actually wrong as flake8 pointed out

Reply:
It's actually fine because I never modify `base_env`,  but have changed to `None` to appease the linter.